src/RefractAPI.cc

```diff
@@ -19,551 +21,537 @@
-    {
-        refract::MemberElement* element = new refract::MemberElement;
+    element->meta().set(SerializeKey::Classes,
+        generate_element<ArrayElement>([](auto& data) { //
+            data.push_back(make_primitive(SerializeKey::User));
+        }));
 
-        refract::ArrayElement* classes = CreateArrayElement(SerializeKey::User);
+    element->get().key(make_primitive(metadata.node->first));
+    element->get().value(make_primitive(metadata.node->second));
 
-        element->meta[SerializeKey::Classes] = classes;
-        element->set(refract::IElement::Create(metadata.node->first), refract::IElement::Create(metadata.node->second));
+    AttachSourceMap(*element, metadata);
 
-        AttachSourceMap(element, metadata);
+    return std::move(element);
+}
 
-        return element;
+std::unique_ptr<IElement> CopyToRefract(const NodeInfo<std::string>& copy)
+{
+    if (copy.node->empty()) {
+        return nullptr;
     }
 
-    refract::IElement* CopyToRefract(const NodeInfo<std::string>& copy)
-    {
-        if (copy.node->empty()) {
-            return NULL;
-        }
+    auto element = PrimitiveToRefract(copy);
+    element->element(SerializeKey::Copy);
 
-        refract::IElement* element = PrimitiveToRefract(copy);
-        element->element(SerializeKey::Copy);
+    return std::move(element);
+}
 
-        return element;
+std::unique_ptr<IElement> ParameterValuesToRefract(
+    const NodeInfo<snowcrash::Parameter>& parameter, ConversionContext& context)
+{
+    auto element = CollectionToRefract<ArrayElement>(
+        MAKE_NODE_INFO(parameter, values), context, LiteralToRefract, SerializeKey::Enum);
+
+    // Add sample value
+    if (!parameter.node->exampleValue.empty()) {
+        auto samples = make_element<ArrayElement>();
+        samples->get().push_back(
+            CreateArrayElement(LiteralToRefract(MAKE_NODE_INFO(parameter, exampleValue), context)));
+        element->attributes().set(SerializeKey::Samples, std::move(samples));
     }
 
-    refract::IElement* ParameterValuesToRefract(
-        const NodeInfo<snowcrash::Parameter>& parameter, ConversionContext& context)
-    {
-        refract::ArrayElement* element = CollectionToRefract<refract::ArrayElement>(
-            MAKE_NODE_INFO(parameter, values), context, LiteralToRefract<std::string>, SerializeKey::Enum);
-
-        // Add sample value
-        if (!parameter.node->exampleValue.empty()) {
-            refract::ArrayElement* samples = new refract::ArrayElement;
-            samples->push_back(
-                CreateArrayElement(LiteralToRefract<std::string>(MAKE_NODE_INFO(parameter, exampleValue), context)));
-            element->attributes[SerializeKey::Samples] = samples;
+    // Add default value
+    if (!parameter.node->defaultValue.empty()) {
+        element->attributes().set(SerializeKey::Default,
+            CreateArrayElement(LiteralToRefract(MAKE_NODE_INFO(parameter, defaultValue), context)));
```

Comment:
You have replace on line 103 `CreateArrayElement()` by `generate_element<ArrayElement>()`
We shoulde decide to replace all of them (not just there), or leave it untouched for this "turn"

Reply:
Agreed. I've already implemented `make_element<ArrayElement>(...)` to construct from child elements, e. g. `make_element<ArrayElement>(foo(), bar())` where `foo()` and `bar` return `std::unique_ptr<IElement>`. I will modify this during cleanup.